internal/orchestrator/orchestrator.go

```diff
@@ -403,204 +424,461 @@ func (or *orchestrator) initDataExchange(ctx context.Context) (err error) {
 		nodeInfo[i] = node.Profile
 	}
 
-	config := dataexchangeConfig.SubSection(dxPlugin)
-	// Migration for explicitly setting the old name ..
-	if dxPlugin == dxfactory.OldFFDXPluginName ||
-		// .. or defaulting to the new name, but without setting the mandatory URL
-		(dxPlugin == dxfactory.NewFFDXPluginName && config.GetString(ffresty.HTTPConfigURL) == "") {
-		// We need to initialize the migration config, and use that if it's set
-		migrationConfig := dataexchangeConfig.SubSection(dxfactory.OldFFDXPluginName)
-		or.dataexchange.InitConfig(migrationConfig)
-		if migrationConfig.GetString(ffresty.HTTPConfigURL) != "" {
-			// TODO: eventually make this fatal
-			log.L(ctx).Warnf("The %s config key has been deprecated. Please use %s instead", coreconfig.OrgIdentityDeprecated, coreconfig.OrgKey)
-			config = migrationConfig
+	if len(plugins) > 0 {
+		for idx, plugin := range plugins {
+			config := dataexchangeConfig.ArrayEntry(idx)
+			err = plugin.Init(ctx, config.SubSection(config.GetString(dataexchange.DataExchangeConfigType)), nodeInfo, &or.bc)
+			if err != nil {
+				return err
+			}
+			name := config.GetString(dataexchange.DataExchangeConfigName)
+			or.dataexchange[name] = plugin
+		}
+	} else {
+		log.L(ctx).Warnf("Your data exchange config uses a deprecated configuration structure - the data exchange configuration has been moved under the 'plugins' section")
+		dxType := deprecatedDataexchangeConfig.GetString(dataexchange.DataExchangeConfigType)
+		plugin, err := dxfactory.GetPlugin(ctx, dxType)
+		if err != nil {
+			return err
+		}
+
+		config := deprecatedDataexchangeConfig.SubSection(dxType)
+		err = plugin.Init(ctx, config, nodeInfo, &or.bc)
+		if err != nil {
+			return err
 		}
 	}
 
-	return or.dataexchange.Init(ctx, config, nodeInfo, &or.bc)
+	return err
 }
 
 func (or *orchestrator) initPlugins(ctx context.Context) (err error) {
-
 	if or.metrics == nil {
 		or.metrics = metrics.NewMetricsManager(ctx)
 	}
 
-	if err = or.initDatabaseCheckPreinit(ctx); err != nil {
-		return err
-	} else if or.preInitMode {
-		return nil
+	if or.databases == nil {
+		var dp []database.Plugin
```

Comment:
nit: no reason to predeclare this, is there?

Reply:
Yeah I've changed it around a few times, but ultimately decided to keep the `initXPlugin` code inside the `or.xplugin == nil` check, instead of outside it like the deprecated plugins. I'll refactor these and other others.